Orange/widgets/data/tests/test_oweditdomain.py

```diff
@@ -221,3 +249,20 @@ def test_discrete_editor(self):
         self.assertEqual(w.name_edit.text(), "")
         self.assertEqual(w.labels_model.get_dict(), {})
         self.assertIs(w.get_data(), None)
+
+    def test_time_editor(self):
+        w = TimeVariableEditor()
+        self.assertIs(w.get_data(), None)
+
+        v = TimeVariable("T")
+        v.attributes.update({"A": 1, "B": "b"})
+        w.set_data(v)
+
+        self.assertEqual(w.name_edit.text(), v.name)
+        self.assertEqual(w.labels_model.get_dict(), v.attributes)
+        self.assertTrue(w.is_same())
+
+        w.set_data(None)
+        self.assertEqual(w.name_edit.text(), "")
+        self.assertEqual(w.labels_model.get_dict(), {})
+        self.assertIs(w.get_data(), None)
```

Comment:
You might want to add some tests that assure that the new variable has the same value for have_date and have_time as source.